### DISCUSSION TITLE
Always index commonly libraries for auto-import

### DISCUSSION THREAD

**Discussion by cdce8p** [microsoft/pylance-release#2688] [Enhancement]
I find myself wanting to use auto-import most often for the same types and modules I need every day. It would be nice if some / all could always be available even if not indexed from another file. Obviously this is highly usage dependent but having those available could be quite useful for a lot of people.

An incomplete list:
```py
from dataclasses import dataclass
from functools import wraps, partial, cached_property
from pathlib import Path
from typing import TypedDict, NamedTuple, overload
from argparse import ArgumentParser

import os
import sys
import asyncio
```

**Comment by jakebailey** [comment]
If these aren't working, that's a bug; I know we've seen these spuriously not work (I can get sys to be auto-imported just fine, sometimes).

**Comment by cdce8p** [comment]
I checked again, this time with `"python.analysis.indexing": true`. Indeed they all seem to work now. Don't know if these should be available even if it's set to `false`. Currently they are not.

Have to admit that I set `indexing` to `false` after I had some performance issues a while back. Not sure though if the reason was only indexing. Could have been other extensions as well. Might give if it another shoot now to see if it's changed.

**Comment by jakebailey** [comment]
I can't recall if we use the bundled stdlib index if indexing is disabled; I think it would be a good step to enable just that (and no FS scanning) for everyone.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
